src/state/ducks/room.js

```diff
@@ -27,19 +28,30 @@ const login = (username, room) => {
   };
 };
 
-const loginRequest = ({roomId, username}) => ({
+const loginRequest = ({ roomId, username }) => ({
   type: ROOM_LOGIN,
   payload: { roomId, username, logingIn: true }
 });
-const loginSuccess = ({roomId, username}) => ({
+
+const loginSuccess = ({ roomId, username }) => ({
   type: ROOM_LOGIN,
   payload: { roomId, username, logedIn: true }
 });
+
 const loginFailure = (error) => ({
   type: ROOM_LOGIN,
   payload: { error: error }
 });
-const logout = () => ({ type: ROOM_LOGOUT });
+
+const logout = () => {
+  return function(dispatch) {
+    janusApi.leaveRoom();
+
+    dispatch({ type: ROOM_LOGOUT });
+
+    history.push('/');
```

Comment:
As said above, this could change if we are able to manage a more react-router-dom way.